src/jit/morph.cpp

```diff
@@ -6647,19 +6639,55 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
     }
     noway_assert(tree->gtOper == GT_IND);
 
-    // Pass down the current mac; if non null we are computing an address
-    GenTree* res = fgMorphSmpOp(tree, mac);
-
-    if (fldOffset == 0 && res->OperGet() == GT_IND)
+    if (fldOffset == 0)
     {
-        GenTree* addr = res->gtOp.gtOp1;
+        GenTree* addr = tree->gtOp.gtOp1;
+
+        // 'addr' may be a GT_COMMA. Skip over any comma nodes
+        addr = addr->gtEffectiveVal();
+
+#ifdef DEBUG
+        if (verbose)
+        {
+            printf("\nBefore calling fgAddFieldSeqForZeroOffset:\n");
```

Comment:
This seems redundant because `fgAddFieldSeqForZeroOffset` also does a "before" dump.

Reply:
OK, I have removed it

Reply:
Actually This will print the  whole tree (and it occurs after the call to `fgMorphSmpOp`) 
It is helpful to see the final version of the tree to verify that the field sequences are correct and non of them went missing during the call to  `fgMorphSmpOp`

Reply:
If you want could change this to always print the final result of `Compiler::fgMorphField`.
That might be better, since the morphing of a GT_FIELD produces a new more complex tree.

Reply:
That makes sense to me

Reply:
Carol after my last change the assert can now be changed to simply check that the type is BYREF or I_IMPL.

    // We expect 'addr' to be an address at this point.
    assert(addr->TypeGet() == TYP_BYREF || addr->TypeGet() == TYP_I_IMPL);

Reply:
That's great!